modules/generator/registry/counter.go

```diff
@@ -126,6 +151,17 @@ func (c *counter) collectMetrics(appender storage.Appender, timeMs int64, extern
 			lb.Set(name, s.labelValues[i])
 		}
 
+		// If we are about to call Append for the first time on a series, we need
+		// to first insert a 0 value to allow Prometheus to start from a non-null
+		// value.
+		if s.isNew() {
+			_, err = appender.Append(0, lb.Labels(nil), timeMs, 0)
```

Comment:
I'm wondering if we need to add a tiny offset to timeMs here. If the 0 sample and the actual value have the same timestamp, Prometheus might reject the second sample because they would overwrite each other.

Reply:
I was also wondering this.  Perhaps just subtracting 100 from the `timeMs` here would be sufficient.

Reply:
I added with adding a 1 second offset to the initial desired value and updated the tests a bit for this.